python/triton/runtime/interpreter.py

```diff
@@ -1051,12 +1051,24 @@ def __init__(self, fn, arg_names, grid):
         self.constexprs = [name for name in arg_names if __annotations__.get(name) == "constexpr"]
 
     def _init_args_hst(self, args_dev, kwargs):
-        args_hst = []
-        for arg in args_dev:
-            if hasattr(arg, "data_ptr"):
-                args_hst.append(arg.cpu())
+        def _to_cpu(arg):
+            if not hasattr(arg, "data_ptr"):
+                return arg
+
+            if 0 in arg.stride():
+                cpu_arg = arg.new_empty(arg.size(), device='cpu')
```

Comment:
`arg.size()` doesn't return you the actual storage size

Reply:
I'll add in the .storage().size() to fix that bug